kpops/components/streams_bootstrap_v3/kafka_app.py

```diff
@@ -0,0 +1,150 @@
+    )
+    labeled_output_topics: dict[str, KafkaTopicStr] = Field(
+        default={}, description=describe_attr("labeled_output_topics", __doc__)
+    )
+    output_topic: KafkaTopicStr | None = Field(
+        default=None,
+        description=describe_attr("output_topic", __doc__),
+        json_schema_extra={},
+    )
+
+    model_config = ConfigDict(extra="allow")
+
+    @pydantic.field_validator("labeled_output_topics", mode="before")
+    @classmethod
+    def deserialize_labeled_output_topics(
+        cls, labeled_output_topics: dict[str, str] | Any
+    ) -> dict[str, KafkaTopic] | Any:
+        if isinstance(labeled_output_topics, dict):
+            return {
+                label: KafkaTopic(name=topic_name)
+                for label, topic_name in labeled_output_topics.items()
+            }
+        return labeled_output_topics
+
+    @pydantic.field_serializer("labeled_output_topics")
+    def serialize_labeled_output_topics(
+        self, labeled_output_topics: dict[str, KafkaTopic]
+    ) -> dict[str, str]:
+        return {label: topic.name for label, topic in labeled_output_topics.items()}
+
+    # TODO(Ivan Yordanov): Currently hacky and potentially unsafe. Find cleaner solution
+    @pydantic.model_serializer(mode="wrap", when_used="always")
+    def serialize_model(
+        self,
+        default_serialize_handler: pydantic.SerializerFunctionWrapHandler,
+        info: pydantic.SerializationInfo,
+    ) -> dict[str, Any]:
+        return exclude_defaults(
+            self, exclude_by_value(default_serialize_handler(self), None)
+        )
+
+
+class KafkaAppValues(HelmAppValues):
+    """Settings specific to Kafka Apps.
+
+    :param kafka: Kafka streams config
+    """
+
+    kafka: KafkaStreamsConfig = Field(
+        default=..., description=describe_attr("streams", __doc__)
+    )
+
+
+class KafkaAppCleaner(Cleaner, StreamsBootstrap, ABC):
+    """Helm app for resetting and cleaning a streams-bootstrap app."""
+
+    from_: None = None
+    to: None = None
+
+    @property
+    @override
+    def helm_chart(self) -> str:
+        raise NotImplementedError
+
+    @override
+    async def clean(self, dry_run: bool) -> None:
+        """Clean an app using a cleanup job.
+
+        :param dry_run: Dry run command
+        """
+        log.info(f"Uninstall old cleanup job for {self.helm_release_name}")
+        await self.destroy(dry_run)
+
+        log.info(f"Init cleanup job for {self.helm_release_name}")
+        await self.deploy(dry_run)
+
+        if not get_config().retain_clean_jobs:
+            log.info(f"Uninstall cleanup job for {self.helm_release_name}")
+            await self.destroy(dry_run)
+
+
+class KafkaAppV3(PipelineComponent, ABC):
+    """Base component for Kafka-based components.
+
+    Producer or streaming apps should inherit from this class.
+
+    :param values: Application-specific settings
+    """
+
+    values: KafkaAppValues = Field(
```

Comment:
What if we remove this from KafkaApp and then we don't need introduce KafkaAppV3? I don't think this is used anywhere directly, only via streams-bootstrap